content/book/chapter_1/Chapter_1.md

```diff
@@ -0,0 +1,213 @@
+
+The answer, at the moment is rather unsettling - the data is stored _in the memory of the webserver_. If the webserver is stopped, all its memory will be freed and the Todo items will be lost forever. You can make this happen by stopping and restarting the webserver. After you've done that, reload the web page and you'll see this:
+
+![screenshot 4](./screenshot4.png)
+
+For sure, we could have written code in the webserver which stored the todo items in files on the disk, or somewhere else. But generally the best thing to do with data you want to store is to use some software specifically designed for storing data. Permanent storage of data - also called "persistence" - is a huge and complex topic, and it is almost certainly to use some datastore for persistence. A popular category of datastore is the _database_, and our app can be configured to use a database.
+
+### Running with a database
+
+Again, see [the project repo](todo:link) for instructions on how to run the app with a database. You will need to install the database we use (called MySQL) onto your computer, then pass details of how to connect to the database to the webserver when you start it.
+
+Once you have those things running together then you can create Todo items using `curl` or your the web page, and you can restart the webserver without losing your data. You can also restart the database - those really do write the data into files on the disk, so we will find the same data still exists after a restart. We might draw the application architecture like this:
+
+![webapp plus database](./webapp2.png)
+
+In this architecture, we can refer to the webserver as being _stateless_ (in the sense that it isn't responsible for any of the application's _state_). This has a number of advantages, in particular it lets us create multiple identical webservers all connecting to the same database. This might let us server more users, and we can be confident in not losing data if a webserver crashes or hardware fails. Having multiple copies of the database is also possible - and is another good reason why we should use a dedicated database.
+
+*Aside:* Most architecture diagrams are just boxes and lines, but databases are usually drawn as cylinders. Why? Because they store data on disks and early disks were cylindrical:
+
+![disk](./disk.png)
+
+This is why Oracle (a database company originally) has buildings of that shape in its HQ:
+
+![Oracle hq](./orahq.png)
+
+(You might recognise the buildings the HQ of Cyberdyne systems in Terminator Genesis, too)
+
+
+## Accessing the application
+
+### Not everyone is using your computer
+
+(so we can't use localhost forever)
+
+### Accessing from the internet
+
+Your computer is probably not directly connected to the internet, but connects through your router or similar at home:
+
+![home server architecture](./webapp3.png)
```

Comment:
This picture and the next have the browser in the wrong place. It should be connecting from the outside through the router.